homeassistant/components/seventeentrack/sensor.py

```diff
@@ -78,32 +60,57 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Configure the platform and add the sensors."""
-
-    session = aiohttp_client.async_get_clientsession(hass)
+    """Initialize 17Track import from config."""
 
-    client = SeventeenTrackClient(session=session)
-
-    try:
-        login_result = await client.profile.login(
-            config[CONF_USERNAME], config[CONF_PASSWORD]
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+    )
+    if (
+        result["type"] == FlowResultType.CREATE_ENTRY
+        or result["reason"] == "already_configured"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            is_fixable=False,
+            breaks_in_ha_version="2024.7.0",
```

Comment:
```suggestion
            breaks_in_ha_version="2024.9.0",
```
Deprecation period is 6 months and this will be introduced in at least 2024.3